proxy/tests/telemetry.rs

```diff
@@ -678,6 +678,156 @@ fn metrics_endpoint_outbound_request_duration() {
+            .run();
+        let metrics = client::http1(proxy.metrics, "localhost");
+        let client = client::new(
+            proxy.outbound,
+            "labeled-set.test.svc.cluster.local"
+        );
+
+        info!("client.get(/)");
+        assert_eq!(client.get("/"), "hello");
+        let scrape = metrics.get("/metrics");
+        // TODO: we can't make more specific assertions about the metrics
+        // besides asserting that both labels are present somewhere in the
+        // scrape, because testing for whole metric lines would depend on
+        // the order in which the labels occur, and we can't depend on hash
+        // map ordering.
+        assert_contains!(scrape, "dst_set_label1=\"foo\"");
+        assert_contains!(scrape, "dst_set_label2=\"bar\"");
+
+    }
+
+    // https://github.com/runconduit/conduit/issues/613
+    #[test]
+    #[cfg_attr(not(feature = "flaky_tests"), ignore)]
+    fn labeled_addr_and_addrset() {
+        let _ = env_logger::try_init();
+
+        info!("running test server");
+        let srv = server::new()
+            .route("/", "hello")
+            .run();
+
+        let ctrl = controller::new()
+            .labeled_destination(
+                "labeled-all.test.svc.cluster.local",
+                srv.addr,
+                HashMap::from_iter(
+                    vec![(String::from("addr_label"), String::from("foo"))]
+                ),
+                HashMap::from_iter(
+                    vec![(String::from("set_label"), String::from("bar"))]
+                ))
+            .run();
+        let proxy = proxy::new()
+            .controller(ctrl)
+            .outbound(srv)
+            .metrics_flush_interval(Duration::from_millis(500))
+            .run();
+        let metrics = client::http1(proxy.metrics, "localhost");
+
+        let client = client::new(
+            proxy.outbound,
+            "labeled-all.test.svc.cluster.local"
+        );
+
+        info!("client.get(/)");
+        assert_eq!(client.get("/"), "hello");
+        let scrape = metrics.get("/metrics");
+        assert_contains!(scrape,
+            "request_duration_ms_count{authority=\"labeled-all.test.svc.cluster.local\",direction=\"outbound\",dst_addr_label=\"foo\",dst_set_label=\"bar\"} 1");
+        assert_contains!(scrape,
+            "response_duration_ms_count{authority=\"labeled-all.test.svc.cluster.local\",direction=\"outbound\",dst_addr_label=\"foo\",dst_set_label=\"bar\",classification=\"success\",status_code=\"200\"} 1");
+        assert_contains!(scrape,
+            "request_total{authority=\"labeled-all.test.svc.cluster.local\",direction=\"outbound\",dst_addr_label=\"foo\",dst_set_label=\"bar\"} 1");
+        assert_contains!(scrape,
+            "response_total{authority=\"labeled-all.test.svc.cluster.local\",direction=\"outbound\",dst_addr_label=\"foo\",dst_set_label=\"bar\",classification=\"success\",status_code=\"200\"} 1");
```

Comment:
In this test, we can make assertions about the contents of whole metric lines, while in the others, we cannot because hashmap ordering is not predictable. Here, since there is only one label on the address set and one label on the address, the labels always appear in a predictable order.

Reply:
I see. IMO, it would be better to just make this test consistent with the others, because it is confusing, especially since status stuff doesn't actually matter in this case (IIUC).